src/utils/format-data.js

```diff
@@ -5,169 +5,109 @@
-    return {};
-  }
-
-  // Snapshots
-  const snapshotIDs = [];
-  const snapshotSchema = {};
-  const snapshotMessage = {};
-  const snapshotTimestamp = {};
-  const snapshotNodes = {};
-  const snapshotEdges = {};
-  const snapshotTags = {};
+const createPipelineState = () => ({
+  id: null,
   // Nodes
-  const nodeID = {};
-  const nodeName = {};
-  const nodeFullName = {};
-  const nodeType = {};
-  const nodeIsParam = {};
-  const nodeTags = {};
+  nodes: [],
+  nodeID: {},
+  nodeName: {},
+  nodeFullName: {},
+  nodeType: {},
+  nodeIsParam: {},
+  nodeTags: {},
+  nodeActive: {},
+  nodeDisabled: {},
+  nodeFocused: null,
   // Edges
-  const edgeSources = {};
-  const edgeTargets = {};
+  edges: [],
+  edgeSources: {},
+  edgeTargets: {},
+  edgeActive: {},
+  edgeDisabled: {},
   // Tags
-  const tagName = {};
-
-  /**
-   * Format raw data for a single snapshot into a usable structure
-   * @param {string} snapshotID - Unique snapshot ID
-   * @param {Object} raw - The parsed data straight from the JSON
-   * @return {Object} The node, edge and raw data for the chart
-   */
-  const formatSnapshotData = (snapshotID, rawSnapshot) => {
-    if (!validateInput(rawSnapshot)) {
-      return;
-    }
-
-    snapshotNodes[snapshotID] = [];
-    snapshotEdges[snapshotID] = [];
-    snapshotTags[snapshotID] = [];
+  tags: [],
+  tagName: {},
+  tagActive: {},
+  tagEnabled: {}
+});
 
-    /**
-     * Add a new node if it doesn't already exist
-     * @param {string} name - Default node name
-     * @param {string} type - 'data' or 'task'
-     * @param {Array} tags - List of associated tags
-     */
-    const addNode = node => {
-      const id = getNodeID(snapshotID, node.id);
-      if (nodeName[id]) {
-        return;
-      }
-      snapshotNodes[snapshotID].push(id);
-      nodeID[id] = id;
-      nodeName[id] = node.name;
-      nodeFullName[id] = node.full_name;
-      nodeType[id] = node.type;
-      nodeIsParam[id] = Boolean(node.is_parameters);
-      nodeTags[id] = (node.tags || []).map(tagID =>
-        getTagID(snapshotID, tagID)
-      );
-    };
-
-    /**
-     * Create a new link between two nodes and add it to the edges array
-     * @param {Object} source - Parent node
-     * @param {Object} target - Child node
-     */
-    const addEdge = ({ source, target }) => {
-      const id = getEdgeID(snapshotID, source, target);
-      if (snapshotEdges[snapshotID].includes(id)) {
-        return;
-      }
-      snapshotEdges[snapshotID].push(id);
-      edgeSources[id] = getNodeID(snapshotID, source);
-      edgeTargets[id] = getNodeID(snapshotID, target);
-    };
+/**
+ * Add a new node if it doesn't already exist
+ * @param {string} name - Default node name
+ * @param {string} type - 'data' or 'task'
+ * @param {Array} tags - List of associated tags
+ */
+const addNode = state => node => {
+  const { id } = node;
+  if (state.nodeName[id]) {
+    return;
+  }
+  state.nodes.push(id);
+  state.nodeID[id] = id;
```

Comment:
Silly question - we have 

```
if (state.nodeName[id]) {
```

That checks if it exists, then we have state.nodeID[id] = id; - are the nodeName and nodeID arrays the same? Why is it passing an ID to state.nodeName[id] rather than something called "name"

Reply:
Is this for convenience for looking it up based on the ID?

Reply:
Ah yeah `state.nodeID` exists because previously I needed a way to get the original node ID, because the one used for lookups was getting the snapshot ID added to it. I might be able to remove this but I'll need to confirm.

nodeName and nodeID are not necessarily the same though.

Reply:
I've removed it, as it doesn't appear to do anything anymore 🤦‍♂